staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -87,6 +93,17 @@ type authProxyRoundTripper struct {
 	rt http.RoundTripper
 }
 
+// NewCacheRoundTripper creates a roundtripper that reads the ETag on
+// response headers and send the If-None-Match header on subsequent
+// corresponding requests.
+func NewCacheRoundTripper(cacheDir string, rt http.RoundTripper) http.RoundTripper {
+	d := diskv.New(diskv.Options{BasePath: cacheDir})
+	t := httpcache.NewTransport(diskcache.NewWithDiskv(d))
```

Comment:
The server can use `cache-control` to let the client (and proxies) know how/when to cache the filed, and when it needs to be downloaded again. As an example, the server can specify `stale-if-error` to say that the client can re-use the latest known version of the resource when the server is broken. If we would use that, then the client would return the version from cache (file hasn't changed and is coming from cache), even though the server is down. That's definitely going to break your use-case.

I think you need a stronger semantic to know if something changed. It'd be more interesting for example, to use the version field of the openapi-spec to verify if the file has changed. The server would have to set this field to a different value based on changes, but that'd be much less flaky.